src/components/SuggestedItem/SuggestedItem.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import "./SuggestedItem.css";
+
+export default props => {
+  const {
+    // flag,
+    handleClick,
+    image,
+    name,
+    pack,
+    index
+  } = props;
+  const toBePacked = pack ? "item--font--packed" : "item--font--unpacked";
+  const borderColor = pack ? " item--border--packed " : " item--border--unpacked ";
+  return (
+    <div
+      className={toBePacked + borderColor +"  item p-0 m-1 rounded"}
+      onClick={handleClick("bundle", index)}
+    >
+      <div className="container">
+        <div className={"row align-items-center justify-content-center"}>
+          <div className="col-12 px-1 pt-1  text-left div--icon--size">
+            {/* ICONS */}
+            {pack ? (
+              <i className="far fa-check-circle align-top item--icon--bundled" />
+            ) : (
+              <i className="far fa-circle align-top item--icon--unbundled" />
+            )}
+          </div>
+          <div>
+            {/* ITEM IMAGE */}
+            <img src={image} className="item--image--size rounded" alt={name} />
+          </div>
+        </div>
+      </div>
+      <div className="col-12">
+        {/* Item Name */}
+        <div className={"row align-items-center justify-content-center"}>
+          <span className="item--title--font">{name}</span>
+        </div>
+      </div>
+      <div className="col-12" onClick={handleClick("select", index)}>
```

Comment:
This looks like a bug 😥

Typing up more now, brb...

Reply:
Ok! So the problem here is that you're passing the _result_ of _calling_ `handleClick` to the `div`'s `onClick` attribute. 

Take this simpler version of your code:

```jsx
const MyComponent = ({ handleClick }) => {
    return <div onClick={handleClick()}>hi</div>;
};
```

That ☝ is the equivalent of this 👇:


```jsx
const MyComponent = ({ handleClick }) => {
    const resultOfHandleClick = handleClick();
    return <div onClick={resultOfHandleClick}>hi</div>;
};
``` 

You want to pass a function that can be called to pass some arguments to `handleClick` ("currying"), but you're actually just immediately calling `handleClick` exactly once.

---

Likely, what you wanted to do was **this** 👇:

```jsx
const MyComponent = ({ handleClick }) => {
    return <div onClick={() => handleClick()}>hi</div>;
};
```

...which is the equivalent of 👇:

```jsx
const MyComponent = ({ handleClick }) => {
    const lambdaThatCallsHandleClick = () => handleClick();
    return <div onClick={lambdaThatCallsHandleClick}>hi</div>;
};
```

Reply:
@FiveEightyEight oh, got it! In that case it looks like you're doing things correctly, just the name `handleClick` confused me. How about switching the name to `createHandleClick` or `createClickHandler`?

Reply:
@JoshuaKGoldberg Thanks Josh! I went ahead and made the change for legibility